### PR TITLE
E-RPM fix

### PR DESCRIPTION
**Pull request by egonl**
Merged https://github.com/mathiasvr/bluejay/releases/tag/erpm-fix2 in main and created a PR in order to build it. See #47.

### CODE REVIEW

**Comment by egonl** [comment]
Here's the build for testing: [test-erpm-fix.zip](https://github.com/bird-sanctuary/bluejay/files/10304913/test-erpm-fix.zip)

**Comment by egonl** [comment]
Quickly tested it and it seems to be as effective as the Bluejay 0.14 ERPM Test release from mathiasvr. Will do some more tests,

image

File: Bluejay.asm [hunk 1]
```diff
@@ -3068,21 +3068,21 @@ dshot_tlm_create_packet:
...
-	mov	Tlm_Data_L, Comm_Period4x_L
+	mov	A, Comm_Period4x_L			; Read commutation period
 	mov	Tlm_Data_H, Comm_Period4x_H
 	setb	IE_EA
 
-	; Multiply period by 3/4 (1/2 + 1/4)
-	mov	A, Tlm_Data_L
+	; Calculate e-period (6 commutations) in microseconds
+	; Comm_Period * 6 * 0.5 = Comm_Period4x * 3/4 (1/2 + 1/4)
+	;mov	A, Tlm_Data_L
```
Comment (stylesuxx) [inline Bluejay.asm:3078]: No need to carry commented out code, please remove completely.

**Comment by stylesuxx** [comment]
I tried to (quantify the changes with a tachometer)[https://docs.google.com/spreadsheets/d/1Lq1zECr-beX1VMak8TCUhBmzwYTZCre37D5LFFnj6PQ/edit?usp=sharing]. Unfortunately this does not seem to be the best way. The tachometer readings are very steady, the period times are a bit all over the place...

@egonl's screenshot makes sense, so do the changes in code. Maybe you could attach a before and after log, just so that we have some hard data to back up the changes.

@damosvil, @saidinesh5 could you maybe have a look over this too and check if you can see any issues?

**Comment by damosvil** [comment]
I have checked the code, and it looks fine. I will also try to check it on my side

File: Bluejay.asm (hunk 1 above)
Comment (egonl) [reply Bluejay.asm]: You're right, removed it.

**Comment by egonl** [comment]
Added two logs of an double brushless XK K130 helicopter, with thr exact the same FC settings. The only difference is the Bluejay version:
- LOG456, with Bluejay v0.16
- LOG457, with Bluejay v0.18 + e-rpm fix (this PR).

Use Rotorflight Blackbox Viewer for opening the files. Betaflight blackbox viewer aso works, but won't always display correct information. If you look at the roll gyro, you'll see that LOG457 is much cleaner as LOG456, i.e. the RPM filters ork better.
[LOG456 - bluejay 0.16.TXT](https://github.com/bird-sanctuary/bluejay/files/10340308/LOG456.-.bluejay.0.16.TXT)
[LOG457 - bluejay 0.18 + erpm fix.TXT](https://github.com/bird-sanctuary/bluejay/files/10340309/LOG457.-.bluejay.0.18.%2B.erpm.fix.TXT)

**Comment by stylesuxx** [comment]
@egonl Perfect - thank you!

**Comment by damosvil** [comment]
I tested it on my side. Looks good

**Pull request merged**